modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/category/facet/portlet/CategoryFacetPortletPreferencesImpl.java

```diff
@@ -25,63 +25,60 @@
  * @author Lino Alves
  */
 public class CategoryFacetPortletPreferencesImpl
-	implements CategoryFacetPortletPreferences {
+	extends BasePortletPreferences implements CategoryFacetPortletPreferences {
 
 	public CategoryFacetPortletPreferencesImpl(
 		Optional<PortletPreferences> portletPreferencesOptional) {
 
-		_portletPreferencesHelper = new PortletPreferencesHelper(
-			portletPreferencesOptional);
+		super(portletPreferencesOptional.get());
```

Comment:
@tinatian wouldn't this throw NoSuchElementException if portletPreferences is not present in the optional?
I thought that's why we added below
`portletPreferencesOptional.orElse(PortletPreferencesFactoryUtil.getEmptyPortletPreferences())`